core/src/dird/ua_select.cc

```diff
@@ -1107,6 +1107,56 @@ void AddPrompt(UaContext* ua, std::string&& prompt)
   AddPrompt(ua, p.c_str());
 }
 
+
+/**
+ * Formats the prompts of a UaContext to be displayed in a multicolumn output
+ * when possible
+ */
+std::string FormatMulticolumnPrompts(const UaContext* ua,
+                                     const int windowsize,
+                                     const int min_lines_threshold)
+{
+  unsigned int max_prompt_length = 1;
+  int prompts_perline = 1;
+  int max_line_length = 1;
+
+  int prompt_max_index_length = std::to_string(ua->num_prompts).length();
+
+  for (int i = 1; i < ua->num_prompts; i++) {
+    if (strlen(ua->prompt[i]) > max_prompt_length) {
+      max_prompt_length = strlen(ua->prompt[i]);
+    }
+  }
+  max_line_length = max_prompt_length + prompt_max_index_length;
+  prompts_perline = (windowsize / max_line_length) - 1;
+  if (prompts_perline == 0) { prompts_perline = 1; }
+
+
+  max_line_length += 4;
+  char buffer[max_line_length];
```

Comment:
pls use a std::vector<char> instead

Reply:
I need this c-string in order to use it within another function (just a few lines below this declaration) that asks for a c-string argument.